kaitaistruct.py

```diff
@@ -367,7 +369,7 @@ def process_rotate_left(data, amount, group_size):
         anti_amount = -amount & mask
 
         r = bytearray(data)
-        for i in range(len(r)):
+        for i in range(len(r)):  # pylint:disable=consider-using-enumerate
```

Comment:
Hmm, using `enumerate()` as pylint suggests actually looks like a good idea. It results in a more Pythonic code and it is apparently faster as well - see my [simple benchmark](https://gist.github.com/generalmimon/31612fa008753ef0943558e1d8104496):

```console
$ python --version
Python 3.9.6

$ python bench-process-rotate-w-enumerate.py
enumerate loop: [7.436, 7.548, 7.598, 7.622, 7.685]
original "for i in range(len(r))" loop: [8.412, 8.423, 8.448, 8.451, 8.465]

$ python bench-process-rotate-w-enumerate.py
original "for i in range(len(r))" loop: [8.179, 8.190, 8.221, 8.330, 8.397]
enumerate loop: [7.460, 7.500, 7.532, 7.539, 7.555]

$ python2 --version
Python 2.7.18

$ python2 bench-process-rotate-w-enumerate.py
('enumerate loop:', '[24.086, 24.571, 24.621]')
('original "for i in range(len(r))" loop:', '[31.043, 31.066, 31.204]')

$ python2 bench-process-rotate-w-enumerate.py
('original "for i in range(len(r))" loop:', '[30.531, 30.922, 31.063]')
('enumerate loop:', '[24.746, 24.912, 25.190]')
```

(I tried both orders of the functions, because sometimes I couldn't shake a feeling from benchmarks in other languages that the execution time of a case that runs first/last is skewed. But specifically [`timeit`](https://docs.python.org/3/library/timeit.html) seems to avoid this issue, because I don't see any bias here and the results are printed to the console at the same time, which makes an impression to me that [`timeit`](https://docs.python.org/3/library/timeit.html) somehow runs both tests at the same time, which sounds like it would really fix the problem.)